src/main/webapp/script.js

```diff
@@ -83,13 +82,29 @@ function schedule() {
   [startDate, endDate] = daterange.split(' - ');
   formContent.append("startDate", startDate.trim());
   formContent.append("endDate", endDate.trim());
+  appendDurations(formContent);
   postData("/schedule", new URLSearchParams(formContent).toString())
       .then(getJsonIfOk)
       .then(updateView)
       .then(refreshCalendar)
       .catch(handleNetworkError);
 }
 
+/**
+ * Appends to the form data the selected values for the duration of each task.
+ */
+function appendDurations(formData) {
+  $("input[name=taskId]").each(
+      function () {
+        if (this.checked) {
+          const durationSelect = $(this).closest(".task-title").next()[0];
```

Comment:
I think this selector is error-prone. What if we added a button before the selection, or wrapped the selection in a tag?
I think it would be better to use an ID instead for selecting, or maybe we could use something like `$(this).children(".mdc-selection")` (https://api.jquery.com/children/)

Reply:
That is correct observation. Traversal using `next()`, `prev()` etc is likely to become broken when HTML structure change.
ID usage would be invalid, as ID should be unique for the whole page, but using a distinctive CSS class for selector seems like the right idea.